src/common/SearchService/SearchService.js

```diff
@@ -0,0 +1,41 @@
+// @ts-check
+import TrieSearch from 'trie-search';
+
+/**
+ * SearchService - Responsible for all the search operations
+*/
+class SearchService{
+    /**
+     * IData - The data that is to be searched into
+     * @typedef {Object[]} IData
+    */
```

Comment:
You can check if your type annotations are compatible here: https://www.typescriptlang.org/play?filetype=js. You'll see an error if you copy/paste your code in this case because based on this type definition, `IData` will represent an Array of data rows while `keywordRowPairs` defined below will be an Array of string-**single row** pairs. So, you might want to include two type definitions here,
```suggestion
    /**
     * IDataRow - A single row of search data
     * @typedef {Object} IDataRow
     */
     
    /**
     * IData - An Array of search data rows
     * @typedef {IDataRow[]} IData
     */   
```
Note that I also omitted the "to be searched into" phrase as `IData` need not be the entirety of the data. For example, the `return` type of the `search` method would also be `IData` and this would only be the subarray of search results.